tasks/ssh_private_key.yaml

```diff
@@ -0,0 +1,18 @@
+- name: ensure ssh private key is set
+  loop: |
+    {%- set private_keys = [] -%}
+    {%- for user_name, user_value in system.users.items() -%}
+      {%- if user_value.private_key is defined -%}
+        {%- set _ = private_keys.append({'user': user_name, 'key': user_value.private_key}) -%}
+      {%- endif -%}
+    {%- endfor -%}
+    {{ private_keys }}
+  loop_control:
+    label: "{{ item.user }}"
+  copy:
+    dest: "~{{ item.user }}/.ssh/id_ed25519"
+    content: "{{ item.key }}"
+    owner: "{{ item.user }}"
+    group: "{{ item.user }}"
+    mode: 0600
```

Comment:
Recommendation: Do not try to use numeric values for `mode`. In some cases Ansible (PyYAML to be exact) reads this in octal instead of decimal mode and the output will not be what you expect. Symbolic mode (`u+rw,go-rwx`) always works and only changes the bits you actually wanted to change. (Generally it is recommended to change only the things you actually want to change and leave the rest as is.)